assert/assert_greater.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+import { format } from "./_format.ts";
+import { AssertionError } from "./assertion_error.ts";
+
+/**
+ * Make an assertion that `actual` is greater than `expected`.
+ * If not then throw.
+ */
+export function assertGreater<T>(actual: T, expected: T, msg?: string) {
```

Comment:
Does using a generic here (with no `extends` or similar) make sense? There are a limited amount of JS primitives that can be compared using `>`/`<` operators. I think we should consider enumerating them.

Reply:
I think `T` without `extends` is fine because comparing behavior can be customized by `valueOf` method of user defined object, like:

```js
console.log({ valueOf() { return 1 } } < { valueOf() { return 2 } }); // => true
console.log({ valueOf() { return 1 } } < { valueOf() { return 0 } }); // => false
```